### PR TITLE
docs: fully featured project

### PR DESCRIPTION
**Pull request by yuhan**
### Summary & Motivation
Revamping https://docs.dagster.io/guides/dagster/example_project to include
1. getting started / set up
2. detailed walkthrough of the example project, with explanation of whys and recommended practices 
~3. recommended project structure~ separated out to https://github.com/dagster-io/dagster/pull/9223

this is a very initial draft - plz feel free to give feedback about anything!

### How I Tested These Changes
preview: https://dagster-git-yuhan-docs-fully-featured-elementl.vercel.app/guides/dagster/example_project

### CODE REVIEW

**Review by sryza — APPROVED** [review]
I think it's worth addressing this comment I left:

> The project is meant to encapsulate multiple separate use cases (training an ML model, activity analysis), so it might make sense to say something like "centered around a contrived organization that wants to do ML and analysis on hacker news user activity data"

Aside from that, I think this looks great

File: docs/content/guides/dagster/example_project.mdx [hunk 1]
```diff
@@ -1,21 +1,132 @@
 ---
-title: Example Dagster Project | Dagster
-description: This guide describes our example Dagster project
+title: Fully Featured Project  | Dagster
+description: This guide walk through a fully featured Dagster project that uses many of Dagster's features.
```
Comment (clairelin135) [inline docs/content/guides/dagster/example_project.mdx:3]:
```suggestion
description: This guide walks through a fully featured Dagster project that uses many of Dagster's features.
```

File: docs/content/guides/dagster/example_project.mdx [hunk 2]
```diff
@@ -1,21 +1,132 @@
...
+src="/images/guides/fully-featured-project/global-asset-lineage.png"
+width={4064}
+height={2488}
+/>
+
+- [`core`](https://github.com/dagster-io/dagster/tree/master/examples/project_fully_featured/project_fully_featured/assets/core): Contains data sets of activity on Hacker News, fetched from the [Hacker News API](https://github.com/HackerNews/API). These are partitioned by hour and updated every hour.
+
+- [`recommender`](https://github.com/dagster-io/dagster/tree/master/examples/project_fully_featured/project_fully_featured/assets/recommender): A machine learning model that recommends stories to specific users based on their comment history, as well as the features and training set used to fit that model. These are dropped and recreated whenever the core assets receive updates.
+
+- [`activity_analytics`](https://github.com/dagster-io/dagster/tree/master/examples/project_fully_featured/project_fully_featured/assets/activity_analytics): Aggregate statistics computed about Hacker News activity. dbt models and a Python model that depends on them. These are dropped and recreated whenever the core assets receive updates.
```
Comment (clairelin135) [inline docs/content/guides/dagster/example_project.mdx:64]:
```suggestion
- [`activity_analytics`](https://github.com/dagster-io/dagster/tree/master/examples/project_fully_featured/project_fully_featured/assets/activity_analytics): Aggregate statistics computed about Hacker News activity represented by dbt models and a Python model that depends on them. These are dropped and recreated whenever the core assets receive updates.
```

File: docs/content/guides/dagster/example_project.mdx [hunk 3]
```diff
@@ -1,21 +1,132 @@
...
+
+### Varying connections or I/O without changing your DAG
+
+[Resources](/concepts/resources) - A resource is an object that models a connection to a (typically) external service. Resources can be shared between assets, and different implementations of resources can be used depending on the environment. In this example, we built [multiple Hacker News API resources](https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/resources/hn_resource.py), all of which have the same interface but different implementations:
+
+- `HNAPIClient` interacts with the real Hacker News API and gets the full data set, which will be used in production.
+- `HNAPISubsampleClient` talks to the real API but subsample the data, which is much faster than the normal implementation and is great for demoing purposes.
```
Comment (clairelin135) [inline docs/content/guides/dagster/example_project.mdx:71]:
```suggestion
- `HNAPISubsampleClient` talks to the real API but subsamples the data, which is much faster than the normal implementation and is great for demoing purposes.
```

File: docs/content/guides/dagster/example_project.mdx [hunk 4]
```diff
@@ -1,21 +1,132 @@
...
+
+By default, it will load for the local deployment. You can toggle deployments by setting the `DAGSTER_DEPLOYMENT` env var to `prod` or `staging`.
+
+## Integrating other tools
+
+Beyond leveraging Dagster core concepts, this project also uses several dagster integration libraries:
 
 - [dagster_dbt](/\_apidocs/libraries/dagster-dbt)
+
+  - Dagster orchestrates dbt alongside other tools, so you can combine dbt with Python, Spark, etc. in a single workflow. This exapmle includes a standalone [`dbt_project`](https://github.com/dagster-io/dagster/tree/master/examples/project_fully_featured/dbt_project), and load dbt models from an existing dbt `manifest.json` file in the dbt project to Dagster assets. It is useful for larger dbt projects as you may not want to recompile the entire dbt project every time you load the Dagster project.
```
Comment (clairelin135) [inline docs/content/guides/dagster/example_project.mdx:114]:
```suggestion
  - Dagster orchestrates dbt alongside other tools, so you can combine dbt with Python, Spark, etc. in a single workflow. This example includes a standalone [`dbt_project`](https://github.com/dagster-io/dagster/tree/master/examples/project_fully_featured/dbt_project), and loads dbt models from an existing dbt `manifest.json` file in the dbt project to Dagster assets. It is useful for larger dbt projects as you may not want to recompile the entire dbt project every time you load the Dagster project.
```

File: docs/content/guides/dagster/example_project.mdx [hunk 5]
```diff
@@ -1,21 +1,132 @@
...
+  - Check out [Using dbt with Dagster](/integrations/dbt) for more recommendations.
+
 - [dagster_aws](/\_apidocs/libraries/dagster-aws)
+
+  - Dagster provides utilities for interfacing with AWS services, including S3, ECS, Redshift, EMR, etc.
+
 - [dagster_slack](/\_apidocs/libraries/dagster-slack)
+
+  - We provide out-of-box support for messaging a given Slack channel. Dagster also comes with a resource for connecting to Slack. Resources are useful for interacting with Slack, as you may want to send messages in production but mock the Slack resource while testing.
```
Comment (clairelin135) [inline docs/content/guides/dagster/example_project.mdx:123]:
```suggestion
  - Dagster provides out-of-the-box support for messaging a given Slack channel, via a resource that connects to Slack. Resources are useful for interacting with Slack, as you may want to send messages in production but mock the Slack resource while testing.
```

**Review by clairelin135 — APPROVED** [review]
Have a couple suggestions, but otherwise this guide is looking great to me!

File: docs/content/guides/dagster/example_project.mdx [hunk 6]
```diff
@@ -1,21 +1,132 @@
...
+- `HNAPISubsampleClient` talks to the real API but subsample the data, which is much faster than the normal implementation and is great for demoing purposes.
+- `HNSnapshotClient` reads from a local snapshot, which is useful for unit testing or environments where the connection isn't available.
 
-## Integrations Used
+The way we model resources helps separate the business logic in code from environments, e.g. you can easily switch resources without changing your pipeline code.
 
-This project uses several dagster integration libraries:
+[I/O managers](/concepts/io-management/io-managers) - An I/O manager is a special kind of resource that handles storing and loading assets. This example includes [a wide range of I/O managers](https://github.com/dagster-io/dagster/tree/master/examples/project_fully_featured/project_fully_featured/resources) such as:
+
+- [`DuckDBPartitionedParquetIOManager`](https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/resources/duckdb_parquet_io_manager.py): It can interact with Spark and dbt without any long-running process. It minimizes setup difficulty and is useful for local development.
```
Comment (jamiedemaria) [inline docs/content/guides/dagster/example_project.mdx:78]:
```suggestion
- [`DuckDBPartitionedParquetIOManager`](https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/resources/duckdb_parquet_io_manager.py) interacts with Spark and dbt without any long-running process. It minimizes setup difficulty and is useful for local development.
```
this flows better imo, but either way is totally fine

File: docs/content/guides/dagster/example_project.mdx [hunk 7]
```diff
@@ -1,21 +1,132 @@
...
+- [`SnowflakeIOManager`](https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/resources/snowflake_io_manager.py): It handles outputs that are either Spark or Pandas DataFrames and write data to a Snowflake table specified by metadata on the relevant <PyObject object="Out" />. The metadata is helpful for observability, especially in production.
```
Comment (jamiedemaria) [inline docs/content/guides/dagster/example_project.mdx:79]:
```suggestion
- [`SnowflakeIOManager`](https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/resources/snowflake_io_manager.py) handles outputs that are either Spark or Pandas DataFrames and write data to a Snowflake table specified by metadata on the relevant <PyObject object="Out" />. The metadata is helpful for observability, especially in production.
```
same as above

File: docs/content/guides/dagster/example_project.mdx [hunk 8]
```diff
@@ -1,21 +1,132 @@
...
+  - We provide out-of-box support for messaging a given Slack channel. Dagster also comes with a resource for connecting to Slack. Resources are useful for interacting with Slack, as you may want to send messages in production but mock the Slack resource while testing.
+
 - [dagster_pyspark](/\_apidocs/libraries/dagster-pyspark)
+
+  - This project builds a [`PartitionedParquetIOManager`](https://github.com/dagster-io/dagster/blob/master/examples/project_fully_featured/project_fully_featured/resources/parquet_io_manager.py) that can take a PySpark DataFrame and store it in Parquet at the given path. It uses <PyObject module="dagster_pyspark" object="pyspark_resource" /> to access to a PySpark SparkSession for executing PySpark code within Dagster.
+  - Besides that, Dagster ops also can perform computations using Spark. Visit [Using Dagster with Spark](/integrations/spark) to learn more.
+
+## Conclusion
+
+Lastly, as time goes on, this guide will be kept up to date, taking advantage of new Dagster features and learnings from the community. If you have anything you'd like to add, or like to see an example of, don't hesitate to reach out!
```
Comment (jamiedemaria) [inline docs/content/guides/dagster/example_project.mdx:132]:
```suggestion
As time goes on, this guide will be kept up to date, taking advantage of new Dagster features and learnings from the community. If you have anything you'd like to add, or an additional example you'd like to see, don't hesitate to reach out!
```
this reads better to me, but also good to keep as is

**Review by jamiedemaria — COMMENTED** [review]
This is great! i think it'll be a really good way to introduce new users to dagster. left a couple comments in the vercel preview and the diff

**Pull request merged**